guides/keras_hub/object_detection_retinanet.py

```diff
@@ -0,0 +1,796 @@
+eval_ds = eval_ds.prefetch(tf.data.AUTOTUNE)
+
+"""
+## Configure RetinaNet Model
+Configure the model with `backbone`, `num_classes` and `preprocessor`.
+Use callbacks for recording logs and saving checkpoints.
+"""
+
+
+def get_callbacks(experiment_path: str):
+    """Creates a list of callbacks for model training.
+
+    Args:
+      experiment_path (str): Path to the experiment directory.
+
+    Returns:
+      List of keras callback instances.
+    """
+    # models_path = os.path.join(experiment_path, "tf_models")
+    tb_logs_path = os.path.join(experiment_path, "logs")
+    ckpt_path = os.path.join(experiment_path, "weights")
+    return [
+        keras.callbacks.BackupAndRestore(ckpt_path, delete_checkpoint=False),
+        keras.callbacks.TensorBoard(
+            tb_logs_path,
+            update_freq=1,
+        ),
+        keras.callbacks.ModelCheckpoint(
+            ckpt_path + "/{epoch:04d}-{val_loss:.2f}.weights.h5",
+            save_best_only=True,
+            save_weights_only=True,
+            verbose=1,
+        ),
+    ]
+
+
+"""
+## Load backbone weights and preprocessor config
+
+Let's use the "retinanet_resnet50_fpn_coco" pretrained weights as the backbone
+model, applying its predefined configuration from the preprocessor of the
+"retinanet_resnet50_fpn_coco" preset.
+Define a RetinaNet object detector model with the backbone and preprocessor
+specified above, and set `num_classes` to 20 to represent the object categories
+from Pascal VOC.
+Finally, compile the model using Mean Absolute Error (MAE) as the box loss.
+"""
+
+backbone = keras_hub.models.Backbone.from_preset("retinanet_resnet50_fpn_coco")
+
+preprocessor = keras_hub.models.RetinaNetObjectDetectorPreprocessor.from_preset(
+    "retinanet_resnet50_fpn_coco"
+)
+model = keras_hub.models.RetinaNetObjectDetector(
+    backbone=backbone, num_classes=len(CLASSES), preprocessor=preprocessor
+)
+model.compile(box_loss=keras.losses.MeanAbsoluteError(reduction="sum"))
+
+"""
+## Train the model
+Now that the object detector model is compiled, let's train it using the
+training and validation data we created earlier.
+For demonstration purposes, we have used a small number of epochs. You can
+increase the number of epochs to achieve better results.
+
+**Note:** The model trained on L4 GPU, while training on T4 it takes
+significant time.
+"""
+
+model.fit(
+    train_ds,
+    epochs=epochs,
+    validation_data=eval_ds,
+    callbacks=get_callbacks("fine_tuning"),
+)
+
+"""
+### Prediction on evaluation data
+Let's predict the model using our evaluation dataset.
+"""
+images, y_true = next(iter(eval_ds.shuffle(50).take(1)))
+y_pred = model.predict(images)
+
+"""
+### Plot the predictions
+"""
+keras.visualization.plot_bounding_box_gallery(
+    images,
+    bounding_box_format=gt_bbox_format,
+    y_true=y_true,
+    y_pred=y_pred,
+    scale=3,
+    rows=2,
+    cols=2,
+    class_mapping=INDEX_TO_CLASS,
+)
+
+"""
+## Custom training object detector
```

Comment:
space after markdown heading (i will stop commenting this, but please go through the rest of the doc)